syntax = "proto3";

message SumRequest {
  double a = 1;
  double b = 2;
}

message SumReply {
  double s = 1;
}

message MultRequest {
  double a = 1;
  double b = 2;
}

message MultReply {
  double p = 1;
}

message GreatestRequest {
  double a = 1;
  double b = 2;
  double c = 3;
}

message GreatestReply {
  double g = 1;
}

message DivRequest {
  double a = 1;
  double b = 2;
}

message DivReply {
  double q = 1;
  double r = 2;
}

service Calculator {
  rpc Sum (SumRequest) returns (SumReply);
  rpc Mult (MultRequest) returns (MultReply);
  rpc Greatest (GreatestRequest) returns (GreatestReply);
  rpc Div (DivRequest) returns (DivReply);
}
